contracts/V1/StateChannelDiamondProxy/FraudProofFacet.sol

```diff
@@ -113,86 +107,48 @@ contract FraudProofFacet is StateChannelCommon {
-        }); 
-
-        if(fraudBlock.stateSnapshotHash == keccak256(abi.encode(outputStateSnapshot))){
-            revert ErrorValidBlockStateTransition();
-        }       
         address signer = StateChannelUtilLibrary.retriveSignerAddress(
             blockInvalidSTProof.invalidBlock.encodedBlock,
             blockInvalidSTProof.invalidBlock.signature
         );
-        return signer;
-    }
 
-    function _handleBlockOutOfGas(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal returns (address) {
-        BlockOutOfGasProof memory blockOutOfGasProof = abi.decode(encodedProof, (BlockOutOfGasProof));
-        bytes memory latestStateStateMachineState = blockOutOfGasProof.latestStateStateMachineState;
-        Block memory fraudBlock = abi.decode(blockOutOfGasProof.invalidBlock.encodedBlock, (Block));
-        
         if(fraudProofVerificationContext.channelId != fraudBlock.transaction.header.channelId) {
             revert ErrorNotSameChannelId();
         }
 
-        bytes memory data = abi.encodeCall(
-            AStateChannelManagerProxy.executeStateTransitionOnState,
-            (
-                fraudProofVerificationContext.channelId,
-                latestStateStateMachineState,
-                fraudBlock.transaction
-            )
-        );
-
-        (bool success, bytes memory returnData) = address(this).delegatecall{gas: getGasLimit()}(data);
-
-        if (success) {
-            revert ErrorValidBlockStateTransition();
+        if(fraudBlock.previousBlockHash != keccak256(abi.encode(previousBlock))){
+            revert ErrorLinkingPreviousBlock();
         }
-        address signer = StateChannelUtilLibrary.retriveSignerAddress(
-            blockOutOfGasProof.invalidBlock.encodedBlock,
-            blockOutOfGasProof.invalidBlock.signature
+        if(previousBlockStateSnapshot.stateMachineStateHash != keccak256(previousStateStateMachineState) && previousBlock.stateSnapshotHash != keccak256(abi.encode(previousBlockStateSnapshot))){
+            revert ErrorInvalidStateSnapshotHash();
+        }
+        (bool isSuccess, bytes memory encodedModifiedState) = AStateChannelManagerProxy(address(this)).executeStateTransitionOnState(
+            fraudProofVerificationContext.channelId,
+            previousStateStateMachineState,
+            fraudBlock.transaction
         );
+        if(!isSuccess){
+            return signer;
+        }
+        StateSnapshot memory newStateSnapshot = StateSnapshot({
+            stateMachineStateHash: keccak256(encodedModifiedState),
+            participants: getStatemachineParticipants(encodedModifiedState),
+            forkCnt: previousBlockStateSnapshot.forkCnt,
+            latestJoinChannelBlockHash: previousBlockStateSnapshot.latestJoinChannelBlockHash,
+            latestExitChannelBlockHash: previousBlockStateSnapshot.latestExitChannelBlockHash,
```

Comment:
Good for now, but the STF needs to return exitChannels[ ] so we can calculate the latestExitChannelBlockHash and totalWithdrawals - feel free to mock it for now or try implementing that too - I'll see tonight when I get back from my meetings to implement that

Reply:
will check the STF and see